tests/src/seaofnodes/son_test.ml

```diff
@@ -45,12 +45,11 @@ let _ =
     @@ SeaOfNodes.Type.Son.bindings graph
   in
   assert (Interpretor.eval_data data == 42) ;
-  (* check that hash-consing works *)
+  (* check that hash-consing works properly, i.e. there is no duplicate data nodes *)
   let nodes = List.map snd (SeaOfNodes.Type.Son.bindings graph) in
   let data_nodes =
     map_option (function SeaOfNodes.Type.Node.Data data -> Some data | _ -> None) nodes
   in
   assert (
-    List.for_all Fun.id
-    @@ List.map (fun a -> List.for_all (fun n -> not (a = n && a != n)) data_nodes) data_nodes ) ;
+    List.for_all (fun a -> List.for_all (fun n -> not (a = n && a != n)) data_nodes) data_nodes );
```

Comment:
Add a note: we assume acyclic graphs for now, then the `a=n` test will always terminate.